docs/contributing/guidelines-for-writing-upgrade.md

```diff
@@ -2,6 +2,8 @@
 
 Keep in mind that upgrade instructions are written for users that do not understand our system well, so more clear they are, more useful they are.
 
+## Introduction
```

Comment:
any rules about linking to PR and that every buletpoint is one PR (or should be)?

Reply:
There is an instruction in `guidelines-for-writing-upgrade.md` that states 

> Each step should have a link to the related pull request and may contain an additional link or links to make instruction more clear.

I believe it is sufficient.